src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java

```diff
@@ -61,4 +64,24 @@ public DiscountResponseVO update(Long id, DiscountRequestVO request) {
     public void deleteById(Long id) {
         discountRepository.deleteById(id);
     }
-}
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void addDiscountToFavorites(Long userId, Long discountId) {
+        Discount discount = discountRepository.findById(discountId)
+                .orElseThrow(() -> new EntityNotFoundException(Discount.class, "id", discountId));
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException(User.class, "id", userId));
+        discount.getUserFavorites().add(user);
+        discountRepository.save(discount);
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void deleteDiscountFromFavorites(Long userId, Long discountId) {
+        Discount discount = discountRepository.findById(discountId)
+                .orElseThrow(() -> new EntityNotFoundException(Discount.class, "id", discountId));
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException(User.class, "id", userId));
+        discount.getUserFavorites().remove(user);
```

Comment:
Does this removal work well? Probably we should add orphanRemoval flag to the mapping. (I'm not sure)

Reply:
works, deleting connections from favorites table